### PR TITLE
Fix #406

### LINKED ISSUE #406: neon new and build failed on Node11.10.1 Win10
Simple Installed Rust 1.34 stable. Node 11.10.1 
Then execute like this
> neon new my_project
Press ^C at any time to quit.
? version 0.1.0
? description
? node entry point lib/index.js
? git repository
? author Mario
? email lijh01@mail.jj.cn
? license MIT
^C终止批处理操作吗(Y/N)? y  // I wait for at least 30 minutes without any response. So I Ctrl C to Quit.

Then I execute like this
> neon build

Then Got Error like this.
   Compiling neon-runtime v0.2.0
error: failed to run custom build command for `neon-runtime v0.2.0`
process didn't exit successfully: `C:\Work\Project\GoKart\CenterServer\Rust\rust_center\native\target\release\build\neon-runtime-b6860bb15546a35d\build-script-build` (exit code: 101)
--- stdout
'Skipping node-gyp installation as part of npm install.'
audited 184 packages in 2.313s
found 1 high severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details
cargo:node_arch=x64

--- stderr
thread 'main' panicked at 'Couldn't find node_root_dir in node-gyp output.', src\libcore\option.rs:1038:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

neon ERR! cargo build failed

Error: cargo build failed
    at Target.<anonymous> (C:\Users\lijh01\AppData\Roaming\npm\node_modules\neon-cli\lib\target.js:121:35)
    at step (C:\Users\lijh01\AppData\Roaming\npm\node_modules\neon-cli\lib\target.js:32:23)
    at Object.next (C:\Users\lijh01\AppData\Roaming\npm\node_modules\neon-cli\lib\target.js:13:53)
    at fulfilled (C:\Users\lijh01\AppData\Roaming\npm\node_modules\neon-cli\lib\target.js:4:58)
    at processTicksAndRejections (internal/process/next_tick.js:81:5)

I'm pretty sure that windows-built-tools is installed successfully.

### PR DESCRIPTION
**Pull request by usagi**
- Fix `neon new` could not be complete in Windows
- Update packages

### CODE REVIEW

File: cli/package.json [hunk 1]
```diff
@@ -14,44 +14,43 @@
...
+    "@types/semver": "^7.1.0",
+    "@types/tmp": "0.1.0",
+    "chai": "^4.2.0",
+    "mocha": "^7.1.1",
+    "nexpect": "^0.6.0",
+    "tmp": "0.1.0",
     "ts-dict": "^0.1.1",
-    "ts-node": "^3.0.2",
+    "ts-node": "^8.7.0",
     "ts-unknown": "^0.2.0",
```
Comment (amilajack) [inline cli/package.json]: Should this be removed?

Comment (usagi) [reply cli/package.json]: Oh, this was my omission. I'll fix it. Thank you!

**Comment by usagi** [comment]
Hmm, https://travis-ci.org/github/neon-bindings/neon/jobs/665380072#L1211 . I'll fix it soon ... maybe.; But, I don't know chalk ecosystem, then I need a time for study it.

**Comment by usagi** [comment]
One problem found, I update the version of `@types/chalk` simply, but it's DEPRECATED already.

**Comment by usagi** [comment]
And also https://yarnpkg.com/package/@types/es6-promise was DEPRECATED.

**Comment by usagi** [comment]
Add these fixups first. And then, I'll solve the error from Chalk.

**Comment by usagi** [comment]
Ahh, I got it with a debugging into the Chalk internals... js is really hard. 😂 

image

from the error message shown:

```javascript
Error: Found extraneous } in Chalk template literal
    at C:\Users\usagi\tmp\z\neon\cli\node_modules\command-line-usage\node_modules\chalk\templates.js:109:11
    at String.replace (<anonymous>)
    at module.exports (C:\Users\usagi\tmp\z\neon\cli\node_modules\command-line-usage\node_modules\chalk\templates.js:99:6)
    at chalkTag (C:\Users\usagi\tmp\z\neon\cli\node_modules\command-line-usage\node_modules\chalk\index.js:221:9)
    at Chalk.chalk.template (C:\Users\usagi\tmp\z\neon\cli\node_modules\command-line-usage\node_modules\chalk\index.js:36:20)
    at chalkFormat (C:\Users\usagi\tmp\z\neon\cli\node_modules\command-line-usage\lib\chalk-format.js:5:12)
    at C:\Users\usagi\tmp\z\neon\cli\node_modules\command-line-usage\lib\section\content.js:40:53
    at Array.map (<anonymous>)
    at getContentLines (C:\Users\usagi\tmp\z\neon\cli\node_modules\command-line-usage\lib\section\content.js:40:28)
    at new ContentSection (C:\Users\usagi\tmp\z\neon\cli\node_modules\command-line-usage\lib\section\content.js:18:18)
```

then, that is here:

image

I'll fix it soon. 👍

**Comment by usagi** [comment]
Hm, maybe I forget some fs.unlink trappings. I'll check the CI error details after hours.

**Comment by usagi** [comment]
Ah.. I got it. I tested used Node.js 13.5.0 only in my local environment and I don't read the older version documents. I'll fixup it soon.

image

**Comment by usagi** [comment]
I prepared the `node`-8.17.0 in local then:

```
error semver@7.1.3: The engine "node" is incompatible with this module. Expected version ">=10". Got "8.17.0"
error Found incompatible module.
```

Ah... I have not an interest for an older versions... and the original motivation is not for updates of dependencies. Hmm... maybe I need time to rethinking.

**Comment by amilajack** [comment]
The issue you're running into can be fixed by downgrading `semver` to `6.3.0` and `@types/semver` to `6.2.1`.

**Pull request closed without merging**